# Project-1
# covid19 small dataset analysis

### Here we will perform data analysis on covid data using sql commands
### The notebook is in the form of question-answer where questions are followed by code to answer 

In [1]:
#if we don't have ipython library then first load and install it and then run this
%load_ext sql 

In [2]:
%sql mysql://root:Pr@nit800@localhost/pranitdb    # To connect sql database

0 rows affected.


[]

In [3]:
%sql show tables # To show tables in the database

 * mysql://root:***@localhost/pranitdb
3 rows affected.


[('covid_data',), ('police_data',), ('udemy_courses',)]

In [4]:
%sql select * from covid_data limit 10 # To show data in the table
# We use 'limit' command if we have to show required no. of rows. In other words we can limit the rows to be displayed

 * mysql://root:***@localhost/pranitdb
10 rows affected.


[('4/29/2020', '', 'Afghanistan', 1939, 60, 252),
 ('4/29/2020', '', 'Albania', 766, 30, 455),
 ('4/29/2020', '', 'Algeria', 3848, 444, 1702),
 ('4/29/2020', '', 'Andorra', 743, 42, 423),
 ('4/29/2020', '', 'Angola', 27, 2, 7),
 ('4/29/2020', '', 'Antigua and Barbuda', 24, 3, 11),
 ('4/29/2020', '', 'Argentina', 4285, 214, 1192),
 ('4/29/2020', '', 'Armenia', 1932, 30, 900),
 ('4/29/2020', '', 'Austria', 15402, 580, 12779),
 ('4/29/2020', '', 'Azerbaijan', 1766, 23, 1267)]

In [5]:
%sql show columns from covid_data  # unlike 'select' ,we have separate 'show' command to show columns from the table

 * mysql://root:***@localhost/pranitdb
6 rows affected.


[('Date', 'text', 'YES', '', None, ''),
 ('State', 'text', 'YES', '', None, ''),
 ('Region', 'text', 'YES', '', None, ''),
 ('Confirmed', 'int', 'YES', '', None, ''),
 ('Deaths', 'int', 'YES', '', None, ''),
 ('Recovered', 'int', 'YES', '', None, '')]

In [6]:
%sql select count(*) from covid_data  # To 'count' total no. of rows in the table

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[(304,)]

### 1) Show the number of Confirmed, Deaths and Recovered cases in each Region.

In [7]:
%sql select Region,Confirmed,Deaths,Recovered from covid_data group by Region limit 10 # 'group by' command group the data in resectice distinct sets

 * mysql://root:***@localhost/pranitdb
10 rows affected.


[('Afghanistan', 1939, 60, 252),
 ('Albania', 766, 30, 455),
 ('Algeria', 3848, 444, 1702),
 ('Andorra', 743, 42, 423),
 ('Angola', 27, 2, 7),
 ('Antigua and Barbuda', 24, 3, 11),
 ('Argentina', 4285, 214, 1192),
 ('Armenia', 1932, 30, 900),
 ('Austria', 15402, 580, 12779),
 ('Azerbaijan', 1766, 23, 1267)]

### 2) Remove all the records where the Confirmed Cases is Less Than 10.

In [8]:
%sql delete from covid_data where Confirmed<10 # We use 'delete' command to delete particular row satisfying condition .#To delete the columns from the table we use 'drop' command'

 * mysql://root:***@localhost/pranitdb
0 rows affected.


[]

In [9]:
%sql select count(*) from covid_data

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[(304,)]

#### Here we can see the total no. of rows have reduced from 321 to 304 as we perform the delete operation. We can verify our given commands by this.

### 3) In which Region, maximum number of Confirmed cases were recorded ?

In [10]:
%sql select region,confirmed from covid_data order by confirmed desc limit 1

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[('US', 299691)]

### Answer: US

### 4) In which Region, minimum number of Deaths cases were recorded ?

In [11]:
%sql select region,deaths from covid_data order by deaths asc limit 50

 * mysql://root:***@localhost/pranitdb
50 rows affected.


[('Eritrea', 0),
 ('France', 0),
 ('Seychelles', 0),
 ('Mainland China', 0),
 ('Mongolia', 0),
 ('Laos', 0),
 ('Mainland China', 0),
 ('Canada', 0),
 ('Australia', 0),
 ('Mozambique', 0),
 ('Mainland China', 0),
 ('Nepal', 0),
 ('Namibia', 0),
 ('France', 0),
 ('Vietnam', 0),
 ('Uganda', 0),
 ('Madagascar', 0),
 ('Grenada', 0),
 ('Canada', 0),
 ('Timor-Leste', 0),
 ('UK', 0),
 ('US', 0),
 ('Canada', 0),
 ('UK', 0),
 ('Denmark', 0),
 ('Rwanda', 0),
 ('Dominica', 0),
 ('South Sudan', 0),
 ('Cambodia', 0),
 ('Macau', 0),
 ('Central African Republic', 0),
 ('France', 0),
 ('Holy See', 0),
 ('Saint Kitts and Nevis', 0),
 ('Saint Lucia', 0),
 ('Denmark', 0),
 ('Fiji', 0),
 ('Saint Vincent and the Grenadines', 0),
 ('Mainland China', 0),
 ('Canada', 0),
 ('Benin', 1),
 ('UK', 1),
 ('Botswana', 1),
 ('Brunei', 1),
 ('Burundi', 1),
 ('Cabo Verde', 1),
 ('Mainland China', 1),
 ('Liechtenstein', 1),
 ('Suriname', 1),
 ('Equatorial Guinea', 1)]

### 5) How many Confirmed, Deaths & Recovered cases were reported from India till 29 April 2020 ?

In [12]:
%sql show columns from covid_data

 * mysql://root:***@localhost/pranitdb
6 rows affected.


[('Date', 'text', 'YES', '', None, ''),
 ('State', 'text', 'YES', '', None, ''),
 ('Region', 'text', 'YES', '', None, ''),
 ('Confirmed', 'int', 'YES', '', None, ''),
 ('Deaths', 'int', 'YES', '', None, ''),
 ('Recovered', 'int', 'YES', '', None, '')]

In [13]:
%sql select Confirmed,Deaths,Recovered from covid_data where region='India' and Date = '4/29/2020' #We must take care of  dat fromat while coding otherwise it would throw error

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[(33062, 1079, 8437)]

#### We can solve this by onother command as below

In [14]:
%sql select * from covid_data where region='india'  # We have 'where' clause to give condition to retrive particlar information from table

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[('4/29/2020', '', 'India', 33062, 1079, 8437)]

### 6) Sort the entire data wrt No. of Confirmed cases in ascending order.

In [15]:
%sql select * from covid_data order by confirmed asc limit 10  # 'order by' clause to arrange data in ascending or descending order 

 * mysql://root:***@localhost/pranitdb
10 rows affected.


[('4/29/2020', '', 'Holy See', 10, 0, 2),
 ('4/29/2020', '', 'Suriname', 10, 1, 8),
 ('4/29/2020', '', 'Gambia', 10, 1, 8),
 ('4/29/2020', 'Greenland', 'Denmark', 11, 0, 11),
 ('4/29/2020', '', 'Burundi', 11, 1, 4),
 ('4/29/2020', 'Yukon', 'Canada', 11, 0, 0),
 ('4/29/2020', '', 'Seychelles', 11, 0, 6),
 ('4/29/2020', 'Montserrat', 'UK', 11, 1, 2),
 ('4/29/2020', 'Turks and Caicos Islands', 'UK', 12, 1, 5),
 ('4/29/2020', 'Falkland Islands (Malvinas)', 'UK', 13, 0, 11)]

### 7) Sort the entire data wrt No. of Recovered cases in descending order.

In [16]:
%sql select * from covid_data order by Recovered desc limit 10

 * mysql://root:***@localhost/pranitdb
10 rows affected.


[('4/29/2020', '', 'Spain', 236899, 24275, 132929),
 ('4/29/2020', '', 'Germany', 161539, 6467, 120400),
 ('4/29/2020', '', 'Iran', 93657, 5957, 73791),
 ('4/29/2020', '', 'Italy', 203591, 27682, 71252),
 ('4/29/2020', 'Hubei', 'Mainland China', 68128, 4512, 63616),
 ('4/29/2020', '', 'France', 165093, 24087, 48228),
 ('4/29/2020', '', 'Turkey', 117589, 3081, 44040),
 ('4/29/2020', '', 'Brazil', 79685, 5513, 34132),
 ('4/29/2020', '', 'Switzerland', 29407, 1716, 22600),
 ('4/29/2020', '', 'Ireland', 20253, 1190, 13386)]

## 8) Find  states  with their confirmed and death cases 

In [17]:
%sql select state, Confirmed,deaths from covid_data group by state

 * mysql://root:***@localhost/pranitdb
130 rows affected.


[('', 1939, 60),
 ('Alabama', 6912, 256),
 ('Alaska', 355, 9),
 ('Alberta', 5165, 87),
 ('Anhui', 991, 6),
 ('Arizona', 7209, 308),
 ('Arkansas', 3193, 57),
 ('Aruba', 100, 2),
 ('Australian Capital Territory', 106, 3),
 ('Beijing', 593, 9),
 ('Bermuda', 111, 6),
 ('British Columbia', 2087, 109),
 ('California', 48747, 1946),
 ('Cayman Islands', 73, 1),
 ('Channel Islands', 537, 38),
 ('Chongqing', 579, 6),
 ('Colorado', 14758, 766),
 ('Connecticut', 26767, 2169),
 ('Curacao', 16, 1),
 ('Delaware', 4655, 144),
 ('Diamond Princess cruise ship', 49, 0),
 ('District of Columbia', 4106, 205),
 ('Falkland Islands (Malvinas)', 13, 0),
 ('Faroe Islands', 187, 0),
 ('Florida', 33193, 1218),
 ('French Guiana', 125, 1),
 ('French Polynesia', 58, 0),
 ('Fujian', 355, 1),
 ('Gansu', 139, 2),
 ('Georgia', 25775, 1101),
 ('Gibraltar', 141, 0),
 ('Grand Princess', 13, 0),
 ('Greenland', 11, 0),
 ('Guadeloupe', 149, 12),
 ('Guam', 141, 5),
 ('Guangdong', 1588, 8),
 ('Guangxi', 254, 2),
 ('Guizhou', 147, 2),
 ('Hainan', 168, 6),
 ('Hawaii', 613, 16),
 ('Hebei', 328, 6),
 ('Heilongjiang', 939, 13),
 ('Henan', 1276, 22),
 ('Hong Kong', 1037, 4),
 ('Hubei', 68128, 4512),
 ('Hunan', 1019, 4),
 ('Idaho', 1952, 60),
 ('Illinois', 50358, 2215),
 ('Indiana', 17182, 964),
 ('Inner Mongolia', 200, 1),
 ('Iowa', 6843, 148),
 ('Isle of Man', 313, 21),
 ('Jiangsu', 653, 0),
 ('Jiangxi', 937, 1),
 ('Jilin', 111, 1),
 ('Kansas', 3839, 134),
 ('Kentucky', 4537, 234),
 ('Liaoning', 146, 2),
 ('Louisiana', 27660, 1845),
 ('Macau', 45, 0),
 ('Maine', 1056, 52),
 ('Manitoba', 275, 6),
 ('Martinique', 175, 14),
 ('Maryland', 20849, 1078),
 ('Massachusetts', 60265, 3405),
 ('Mayotte', 460, 4),
 ('Michigan', 40399, 3670),
 ('Minnesota', 4644, 319),
 ('Mississippi', 6569, 250),
 ('Missouri', 7660, 338),
 ('Montana', 451, 16),
 ('Montserrat', 11, 1),
 ('Nebraska', 3851, 56),
 ('Nevada', 4934, 230),
 ('New Brunswick', 118, 0),
 ('New Caledonia', 18, 0),
 ('New Hampshire', 2058, 60),
 ('New Jersey', 116365, 6771),
 ('New Mexico', 3213, 112),
 ('New South Wales', 3016, 40),
 ('New York', 299691, 23477),
 ('Newfoundland and Labrador', 258, 3),
 ('Ningxia', 75, 0),
 ('North Carolina', 10180, 382),
 ('North Dakota', 1033, 19),
 ('Northern Mariana Islands', 14, 2),
 ('Northern Territory', 28, 0),
 ('Nova Scotia', 935, 28),
 ('Ohio', 17303, 937),
 ('Oklahoma', 3473, 214),
 ('Ontario', 16978, 1153),
 ('Oregon', 2446, 101),
 ('Pennsylvania', 46327, 2373),
 ('Prince Edward Island', 27, 0),
 ('Puerto Rico', 1433, 86),
 ('Qinghai', 18, 0),
 ('Quebec', 26610, 1762),
 ('Queensland', 1033, 6),
 ('Reunion', 420, 0),
 ('Rhode Island', 8247, 251),
 ('Saskatchewan', 383, 6),
 ('Shaanxi', 306, 3),
 ('Shandong', 787, 7),
 ('Shanghai', 647, 7),
 ('Shanxi', 197, 0),
 ('Sichuan', 561, 3),
 ('Sint Maarten', 75, 13),
 ('South Australia', 438, 4),
 ('South Carolina', 5882, 231),
 ('South Dakota', 2373, 13),
 ('St Martin', 38, 3),
 ('Tasmania', 219, 12),
 ('Tennessee', 10366, 195),
 ('Texas', 27257, 754),
 ('Tianjin', 190, 3),
 ('Turks and Caicos Islands', 12, 1),
 ('Utah', 4497, 45),
 ('Vermont', 862, 47),
 ('Victoria', 1361, 18),
 ('Virgin Islands', 57, 4),
 ('Virginia', 14962, 522),
 ('Washington', 14070, 801),
 ('West Virginia', 1110, 38),
 ('Western Australia', 551, 8),
 ('Wisconsin', 6520, 308),
 ('Wyoming', 545, 7),
 ('Xinjiang', 76, 3),
 ('Yukon', 11, 0),
 ('Yunnan', 185, 2),
 ('Zhejiang', 1268, 1)]

## 9) Find how many confirmed cases are reported in the year 2020

In [18]:
%sql select count(confirmed) from covid_data where date like '%2020%'

 * mysql://root:***@localhost/pranitdb
1 rows affected.


[(304,)]